### DISCUSSION TITLE
Displaying external software systems under software systems menu section

### DISCUSSION THREAD

**Discussion by kristoflagae** [avisi-cloud/structurizr-site-generatr#418] [Ideas]
Hello,

This is my first discussion item here...

Is there a reason external software systems are not displayed under the 'software systems' section on the left?

All internal and external software systems are displayed on the 'software systems' page, with their name and a description.
If it's an external system (because of applying the 'external' tag), it gets the 'external' postfix and is displayed in grey text, which is fine.

Within the previous version 1.1.7, this didn't work anymore but it is now fixed again in version 1.1.8.

Because of that previous bug, the external systems were also displayed on the left menu, which in my case looked very handy. 
I continued by creating 2 groups 'Internal systems' and 'External systems'. I also started to create a separate document for each external system with some more details about that external system, which in some cases is much better than just having a description.

But working with the latest version now, these are all gone again.

I would like to have the possibility to see the external systems on the left menu, so I can navigate to their documentation.

Is this also something you find interesting to have, or are there any other ideas?

I know there is a workaround by working with another tag name and changing my related style definition, but then the external postfix will not be displayed anymore on the software systems overview page.

Or would it be an idea to have an option to also make use of a similar grouping usage on the software systems overview page, just like there is the option `generatr.site.nestGroups`  to have a treeview style menu on the left?

**Comment by jp7677** [comment]
I guess you have set the `generatr.site.externalTag` setting to `External systems`? If you still want to include the systems in this group, you could leave out that setting and change the appearance using tags and styles.

**Reply by kristoflagae** [reply]
Correct, that's what I tried to describe as a workaround.

The drawback with that is that on the software systems overview page (https://avisi-cloud.github.io/structurizr-site-generatr/main/software-systems/),  the external systems will not be displayed/postfixed as external or displayed in a separate group (based on my used groups)

**Reply by jp7677** [reply]
Yeah sorry, it is indeed one or the other.

Treating systems outside of the (now deprecated) enterprise boundary like this was a design decision in the beginning of this project, mostly to keep the focus on the "internal" systems and their relations. Especially in a bigger model with lots of external systems, we figured that for our use cases (with not much info linked to the external systems) the menu would contain otherwise a lot of unused items.

Adding a configuration options for also displaying external items in the side menu (with default to `false`) is certainly doable, do you want to give it a try?

**Comment by qtzar** [comment]
The solution I use here is put all the external systems in to their own top level group. In your dsl we have two GROUP lines at the same level, one for the enterprise and one for External. Then with nested groups in the nav turned on it is easy to see.

We don't have the generatr.site.externalTag config setup but we do tag all external systems with a standard tag and have a style setup to reflect a different color for that tag.
